frame/chaos/src/lib.rs

```diff
@@ -0,0 +1,501 @@
+	pub fn EarlyUnstakePenaltyOnEmpty<T: Config>() -> Permill {
+		Zero::zero()
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn early_unstake_penalty)]
+	pub type EarlyUnstakePenalty<T: Config> =
+		StorageValue<_, Permill, ValueQuery, EarlyUnstakePenaltyOnEmpty<T>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn reward_indexes)]
+	pub type RewardIndexes<T: Config> =
+		StorageMap<_, Blake2_128Concat, AssetIdOf<T>, RewardIndex, OptionQuery>;
+
+	#[pallet::type_value]
+	pub fn RewardAssetsOnEmpty<T: Config>() -> BoundedBTreeSet<AssetIdOf<T>, MaxRewardAssetsOf<T>> {
+		BoundedBTreeSet::<AssetIdOf<T>, MaxRewardAssetsOf<T>>::new()
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn reward_assets)]
+	pub type RewardAssets<T: Config> = StorageValue<
+		_,
+		BoundedBTreeSet<AssetIdOf<T>, MaxRewardAssetsOf<T>>,
+		ValueQuery,
+		RewardAssetsOnEmpty<T>,
+	>;
+
+	#[pallet::type_value]
+	pub fn StakingPresetsOnEmpty<T: Config>() -> BoundedBTreeSet<Timestamp, MaxStakingPresetsOf<T>>
+	{
+		BoundedBTreeSet::<Timestamp, MaxStakingPresetsOf<T>>::new()
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn staking_presets)]
+	pub type StakingPresets<T: Config> = StorageValue<
+		_,
+		BoundedBTreeSet<Timestamp, MaxStakingPresetsOf<T>>,
+		ValueQuery,
+		StakingPresetsOnEmpty<T>,
+	>;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Set enabled reward assets.
+		///
+		/// Arguments
+		///
+		/// * `origin` the origin that signed this extrinsic, must be `T::AdminOrigin`.
+		/// * `reward_assets` the set of enabled reward assets.
+		#[pallet::weight(10_000)]
+		pub fn set_reward_assets(
+			origin: OriginFor<T>,
+			reward_assets: BoundedBTreeSet<AssetIdOf<T>, MaxRewardAssetsOf<T>>,
+		) -> DispatchResultWithPostInfo {
+			let _ = T::AdminOrigin::ensure_origin(origin)?;
+			RewardAssets::<T>::set(reward_assets);
+			Ok(().into())
+		}
+
+		/// Set enabled staking presets.
+		///
+		/// Arguments
+		///
+		/// * `origin` the origin that signed this extrinsic, must be `T::AdminOrigin`.
+		/// * `staking_presets` the set of enabled staking presets.
+		#[pallet::weight(10_000)]
+		pub fn set_staking_presets(
+			origin: OriginFor<T>,
+			staking_presets: BoundedBTreeSet<Timestamp, MaxStakingPresetsOf<T>>,
+		) -> DispatchResultWithPostInfo {
+			let _ = T::AdminOrigin::ensure_origin(origin)?;
+			StakingPresets::<T>::set(staking_presets);
+			Ok(().into())
+		}
+
+		/// Overwrite the unlock penalty.
+		/// The penalty will be cut off the stake asset, Chaos in this case.
+		///
+		/// Arguments
+		///
+		/// * `origin` the origin that signed this extrinsic, must be `T::AdminOrigin`.
+		/// * `penalty` the penalty applied to users unstaking early.
+		#[pallet::weight(10_000)]
+		pub fn set_unlock_penalty(
+			origin: OriginFor<T>,
+			penalty: Permill,
+		) -> DispatchResultWithPostInfo {
+			let _ = T::AdminOrigin::ensure_origin(origin)?;
```

Comment:
better name GovernanceOrigin as I read it was dicussed that it will have some own governance